src/Missings.jl

```diff
@@ -207,4 +208,177 @@ missing
 """
 passmissing(f) = PassMissing{Core.Typeof(f)}(f)
 
-end # module
+
+"""
+   skipmissings(args...)
+
+
+Returns a tuple of iterators iterating over indices of elements `args`
+where all elements of `args` are not missing.
+
+# Examples
+```
+julia> x = [1, 2, missing, 4]; y = [1, 2, 3, missing];
+
+julia> tx, ty = skipmissings(x, y)
+(Missings.SkipMissings{Array{Union{Missing, Int64},1}}(Union{Missing, Int64}[1, 2, missing, 
+4], Base.Iterators.Zip{Tuple{Array{Union{Missing, Int64},1},Array{Union{Missing, Int64},1}}}
+((Union{Missing, Int64}[1, 2, missing, 4], Union{Missing, Int64}[1, 2, 3, missing])), 1), 
+Missings.SkipMissings{Array{Union{Missing, Int64},1}}(Union{Missing, Int64}[1, 2, 3, 
+missing], Base.Iterators.Zip{Tuple{Array{Union{Missing, Int64},1},Array{Union{Missing,
+ Int64},1}}}((Union{Missing, Int64}[1, 2, missing, 4], Union{Missing, Int64}[1, 2, 3, 
+ missing])), 2))
+
+julia> collect(tx)
+2-element Array{Any,1}:
+ 1
+ 2
+
+```
+"""
+function skipmissings(args...)
+    z = zip(args...)
+    return ntuple(i -> SkipMissings{typeof(z), i}(z, i), length(args))
+end
+
+struct SkipMissings{C, I}
+    z::C
+    k::Int
+end
+_inner_type(C, I) = C.parameters[1].parameters[I]
+_inner_itr(S::SkipMissings) = S.z.is[S.k]
+_width(itr::SkipMissings) = length(first(itr.z))
+_anymissing(itr::SkipMissings, i) = any(x -> x === missing, (y[i] for y in itr.z.is))
```

Comment:
I think the compiler will optimize this much better. Have you done some benchmarking?
```suggestion
_anymissing(itr::SkipMissings, i) = any(x -> x[i] === missing, itr.z.is)
```

Reply:
Don't forget this when benchmarking as it could make a difference.